lodash.js

```diff
@@ -3699,8 +3699,12 @@
       return function(number, precision) {
         precision = precision ? toInteger(precision) : 0;
         if (precision) {
-          precision = pow(10, precision);
-          return func(number * precision) / precision;
+          // Shift
+          number = number.toString().split('e');
+          number = func(+(number[0] + 'e' + (number[1] ? (+number[1] + precision) : precision)));
+          // Shift back
+          number = number.toString().split('e');
+          return +(number[0] + 'e' + (number[1] ? (+number[1] - precision) : -precision));
```

Comment:
This looks too verbose and formal for lodash... I was planning to compress the logic, e.g.:
```js
// Shift the decimal point with exponential operations
number = (number + 'e').split('e');	// force a non-trivial split
number = func(number[0] + 'e' + (+number[1] + precision));	// rely on ToNumber conversions
// Shift back
number = (number + 'e').split('e');
return +(number[0] + 'e' + (number[1] - precision));
```